Add FIMS user setup bash script (#1067)
NOAA-FIMS/FIMS: .devcontainer/README.md

Reviewer: I am not sure if this is best but we have been formatting all .md files to NOT wrap lines even though it is not best practice within the LaTeX community. The choice mainly goes back to how the lines are displayed on a webbrowser, where people are most likely to interact with them. If the width of their browser is not conducive to 80 characters then the hard break will also lead to a soft break strictly because of their browser width. If we do not wrap lines, then the browser is able to create a line break wherever it is needed. I think this has become more important as accessibility has increased and as we view more websites on tablets and phones. So, that is a long-winded request to not wrap paragraphs at 80 characters in this file 😁.

### Connect VS Code to WSL

- Open VS Code and press `Ctrl + Shift + P` to open the command palette.
- Type and select `WSL: Connect to WSL using Distro`.
- Choose `Ubuntu` from the list.

### Clone FIMS repository (first time only)

- Once connected to WSL, open a new terminal in VS Code.
- Clone the project repository inside the WSL2 file system.
```bash
git clone https://github.com/NOAA-FIMS/FIMS.git
```

> **Note:** For other FIMS-related projects, feel free to use the [example](https://github.com/NOAA-FIMS/FIMS/tree/main/.devcontainer/user) in the `FIMS/.devcontainer/user` folder as a template to set up similar devcontainer environments for your own repositories.

### Open FIMS in Dev Container

- In VS Code and press `Ctrl + O` to open the FIMS Folder you just cloned. 
- Press `Ctrl + Shift + P` to open the command palette.
- Type and select `Dev Containers: Reopen in Container`.
- Select `FIMS User` if you want to run stock assessments using FIMS. Select `FIMS Developer` if you are contributing to the source code or maintaining the FIMS repository.

> - Docker will now build the container. The first build can take several minutes for a `FIMS developer` setup and about 15 minutes for a `FIMS user` setup.

Reviewer: ```suggestion
- Docker will now build the container. The first build can take several minutes for a `FIMS Developer` setup and about 15 minutes for a `FIMS User` setup.
```
changes case of Developer and User to title case

### Close the container

When you're finished, open the Command Palette (`Ctrl + Shift +P`) and type `Remote: Close Remote Connection`.

## 🙋 Need help?

> If any of the instructions above do not work as expected, please [report an issue on GitHub](https://github.com/NOAA-FIMS/FIMS/issues).

Reviewer: I think we should also create a GitHub Discussion about using containers/bash scripts/images and link it here as an additional resource.

- Click the green `Code` button on the [FIMS repository](https://github.com/NOAA-FIMS/FIMS).
- Click the `...` (three dots) next to the `+` icon in the Codespaces tab.
- Select `New with options ...`.
- A dropdown labeled `Dev container configuration` will appear.
- Select `FIMS User` if you want to run stock assessments using FIMS. Select `FIMS Developer` if you are contributing to the source code or maintaining the FIMS repository.
- If you plan to run a FIMS model, change the `Machine type` from the default `2-core` to at least `4-core`. The default will not allow you to compile {FIMS} and will lead to your R session unexpectedly crashing but it is set as the default because we pay for each minute used in Codespaces whether the session is actively being used or sitting idle and 2 cores is much cheaper.
- Click the green `Create codespace` button. 

> > **Note:** The first build can take several minutes for a `FIMS developer` setup and about 15–20 minutes for a `FIMS user` setup.

Author: Thank you, @k-doering-NOAA. I did notice that the pre-build is available when spinning up a new Codespace on this branch, but it still takes ~15 minutes to run the bash script as a post-create command, rather than opening instantly using the prebuild. Does this happen for you as well? If so, should I change `"postCreateCommand": "bash setup_fims.sh"` to `"onCreateCommand": "bash setup_fims.sh"`?

> > **Note:** The first build can take several minutes for a `FIMS developer` setup and about 15–20 minutes for a `FIMS user` setup.

Author: @k-doering-NOAA Just confirmed that using prebuild with `"onCreateCommand": "bash setup_fims.sh"` in `devcontainer.json` speeds everything up. I can now spin up a new Codespace in seconds.

# :package: FIMS devcontainer cheat sheet

This cheat sheet outlines the steps for setting up a consistent FIMS environment.
The approach bypasses manual dependency installation and keeps everyone's setup the same.
Please follow the section that matches how you plan to open the container.

> - :globe_with_meridians: Open the container using a Codescpace from the browser

Reviewer: ```suggestion
- :globe_with_meridians: Open the container using a Codespace from the browser
```

- :technologist: Open the container on a Windows machine using Windows Subsystem

>   for Linux 2 (WSL2), Docker Engine, and Visual Studio Code (VS Code)

Reviewer: I prefer the list of options be included in the sentence rather than as bullets. For example,
> Please follow the section below that matches how you plan to open the container, i.e., follow the [:globe_with_meridians: browser section](#-globe_with_meridians-browser) if you wish to open the container using Codespace from your favorite web browser and [:technologist: local section](#-technologist-local) if you wish to launch the container on your own machine.

I used hyperlinks so users can just click on the appropriate section for them and be taken to the correct location. Also, if you choose to leave this as is, there is a typo in Codespace on line 7.

## :globe_with_meridians: Open the container using a Codescpace from the browser

> - Click the green `Code` button on the FIMS repo.

Author: Done.

- Click the `...` (three dots) next to the `+` icon in the Codespaces tab.
- Select `New with options ...`.
- A dropdown labeled `Dev container configuration` will appear.
- Select `FIMS User` or `FIMS Developer`, depending on your role.
- If you plan to run a FIMS model, change the `Machine type` from the default `2-core` to `16-core`.
- Click the green `Create codespace` button. 

> ## :technologist: Open the container on a Windows machine using WSL2, Docker Engine, and VS Code

Reviewer: ```suggestion
## :technologist: Local

To open the container on your local Windows machine using Windows Subsystem for Linux 2 (WSL2), Docker Engine, and Visual Studio Code (VS Code) use the following instructions:
```
